apps/total-typescript/src/server/get-content-rules.ts

```diff
@@ -0,0 +1,58 @@
+import {z} from 'zod'
+import {NextApiRequest} from 'next'
+import {getToken} from 'next-auth/jwt'
+import {getSubscriberFromCookie} from '../utils/get-convertkit-subscriber-from-cookie'
+import {getWorkshop} from '../lib/workshops'
+import {getTutorial} from '../lib/tutorials'
+import {getExercise} from '../lib/exercises'
+import {getSection} from '../lib/sections'
+import {defineRulesForPurchases, UserSchema} from '../ability/ability'
+
+export const ContentRulesSchema = z.object({
+  moduleSlug: z.string().optional(),
+  moduleType: z.string().optional(),
+  lessonSlug: z.string().optional(),
+  sectionSlug: z.string().optional(),
+  isSolution: z.boolean().optional(),
+  muxPlaybackId: z.nullable(z.string().optional()),
+})
```

Comment:
since all the properties are `optional`, another way to write this is with `partial()`:

```suggestion
export const ContentRulesSchema = z.object({
  moduleSlug: z.string(),
  moduleType: z.string(),
  lessonSlug: z.string(),
  sectionSlug: z.string(),
  isSolution: z.boolean(),
  muxPlaybackId: z.string().nullable(),
}).partial()
```